src/merkle/merkle-objects.ts

```diff
@@ -202,10 +202,13 @@ export class IpfsMerge implements MergeFunction<CIDHolder, TreeMetadata> {
 
   async merge(
     left: Node<CIDHolder>,
-    right: Node<CIDHolder>,
+    right: Node<CIDHolder> | null,
     metadata: TreeMetadata | null
   ): Promise<Node<CIDHolder>> {
-    const merged = [left.data.cid, right.data.cid]
+    const merged = [left.data.cid]
+
+    merged.push(right === null ? null : right.data.cid)
```

Comment:
how did this work if there were an odd number of requests in a batch?  Wouldn't there already be cases where a node had one real child and one null child?

Reply:
So if there were three requests 

left + middle merge creating A
right would not be touched
A + right + metadata would be merged into root

Reply:
ah I see, whenever there's an imbalance we roll the child up and have it replace the parent